### PR TITLE
Add in `VERCEL_ENV` for `vc deploy`

### PR DESCRIPTION
**Pull request by andymckay**
It would be nice in vc pull if an environment variable could be used. So that it can be done once and is consistent for multiple commands. This adds in `VERCEL_ENV` as a environment variable.

For other commands you can use the `--environment` variable. For `deploy` it assumed to be preview unless you specify `--prod`.

This again has `VERCEL_ENV` as a fallback, that is overridden by setting the `--prod` argument.

- [x] `--prebuilt` support
- [x] support when `--prebuilt` is not specified

### CODE REVIEW

File: packages/cli/src/util/deploy/parse-target.ts [hunk 1]
```diff
@@ -37,5 +37,12 @@ export default function parseTarget(
     return 'production';
   }
 
+  if (process.env.VERCEL_ENV) {
+    output.debug(
+      `Setting target to ${process.env.VERCEL_ENV} using VERCEL_ENV environment variable.`
+    );
+    return process.env.VERCEL_ENV;
```
Comment (styfle) [inline packages/cli/src/util/deploy/parse-target.ts]: Should this env var be validated first?

Comment (andymckay) [reply packages/cli/src/util/deploy/parse-target.ts]: Yep, thanks. It's in a later commit 👇🏾 😄

File: packages/cli/src/util/deploy/parse-target.ts [hunk 2]
```diff
@@ -37,5 +37,25 @@ export default function parseTarget(
     return 'production';
   }
 
+  const VERCEL_ENV = process.env.VERCEL_ENV;
+  if (VERCEL_ENV) {
+    if (!['preview', 'staging', 'production'].includes(VERCEL_ENV)) {
```
Comment (styfle) [inline packages/cli/src/util/deploy/parse-target.ts:42]: I don't think we should allow `staging` here. 

Also cc @lucleray who knows more about this target.

**Comment by EndangeredMassa** [comment]
Closing this for now. If anyone needs these, please open a [discussion](https://github.com/vercel/vercel/discussions).

**Pull request closed without merging**